### PR TITLE
[Docs] Document Prompt Behavior with+/out Broker

### PR DESCRIPTION
**Pull request by iambmelt**
Introduces new documentation covering ADAL + MSAL; how `Prompt` augments flows, and OpenId param equivalents

Thanks @kreedula for providing input as to structure/contents of this doc

### Bugs filed
During the writing of this doc, the following issues were discovered and subsequently filed:
- https://github.com/AzureAD/ad-accounts-for-android/issues/1294
- [VSO/Engineering: PBI: 1017217](https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1017217)

### CODE REVIEW

File: docs/prompt_behavior.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+| MSAL    | `WHEN_REQUIRED`   | If `login_hint` is supplied, no `prompt` query parameter will be sent to the `/authorize` endpoint. The user _may_ be preompted to login or grant consent as required by the service/application. If no `login_hint` is supplied, `prompt=select_account`. | If `login_hint` is supplied, unsent <br/> If no `login_hint` is supplied, sends `select_account` | N/A                  | Same as embedded flow                                                                                                                                                                                                                                                                                                  | (unsent - account selection via `AccountChooserActivity`) |
+
+>Note: `CONSENT` not implemented inside of broker with MSAL
```
Comment (iambmelt) [inline docs/prompt_behavior.md:16]: File a bug

Comment (iambmelt) [reply docs/prompt_behavior.md:16]: Make table reflect _desired_ behavior once bug is filed

Comment (iambmelt) [reply docs/prompt_behavior.md:16]: Bug filed here:
https://github.com/AzureAD/ad-accounts-for-android/issues/1294

Contains link to VSO PBI

**Comment by iambmelt** [comment]
CI is failing intermittently, unrelated to these docs changes. I am not planning on investigating, as is unrelated to this PR. @shahzaibj any clue what's up?

For now, I've rerun the pipeline to see if that unjams

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 's'.
> Could not resolve all artifacts for configuration ':classpath'.
   > Could not download tracker.jar (com.android.tools.analytics-library:tracker:26.3.1)
      > Could not get resource 'https://dl.google.com/dl/android/maven2/com/android/tools/analytics-library/tracker/26.3.1/tracker-26.3.1.jar'.
         > Could not GET 'https://dl.google.com/dl/android/maven2/com/android/tools/analytics-library/tracker/26.3.1/tracker-26.3.1.jar'.
            > Read timed out
```

**Comment by shahzaibj** [comment]
> CI is failing intermittently, unrelated to these docs changes. I am not planning on investigating, as is unrelated to this PR. @shahzaibj any clue what's up?
> 
> For now, I've rerun the pipeline to see if that unjams
> 
> ```
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> A problem occurred configuring root project 's'.
> > Could not resolve all artifacts for configuration ':classpath'.
>    > Could not download tracker.jar (com.android.tools.analytics-library:tracker:26.3.1)
>       > Could not get resource 'https://dl.google.com/dl/android/maven2/com/android/tools/analytics-library/tracker/26.3.1/tracker-26.3.1.jar'.
>          > Could not GET 'https://dl.google.com/dl/android/maven2/com/android/tools/analytics-library/tracker/26.3.1/tracker-26.3.1.jar'.
>             > Read timed out
> ```

No sure @iambmelt , but I am seeing a few different errors including a `Service Unavailable` so looks like Maven is down. I've seen this happen a few times before as well, and usually gets resolved a few minutes later.

File: docs/prompt_behavior.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+| MSAL    | `CONSENT`         | Sends query param `prompt=consent` to `/authorize` endpoint. The user will be prompted to grant consent, even if it was previously granted before. `// TODO what happens for pre-authz apps?`                                                              | `consent`                                                                                        | Yes                  |  Sends query param `prompt=consent` to `/authorize` endpoint. The user will be prompted to grant consent, even if it was previously granted before.                                                                                                                                                                                                                                      | `consent`                                                 |
```
Comment (iambmelt) [inline docs/prompt_behavior.md:13]: Got the update from PM team: PreAuthZ apps simply ignore this param

Comment (iambmelt) [reply docs/prompt_behavior.md]: Updated in d12237313c27dc75f4c3b5a863113b49bcdd09f4

File: docs/prompt_behavior.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+| MSAL    | `WHEN_REQUIRED`   | If `login_hint` is supplied, no `prompt` query parameter will be sent to the `/authorize` endpoint. The user _may_ be preompted to login or grant consent as required by the service/application. If no `login_hint` is supplied, `prompt=select_account`. | If `login_hint` is supplied, unsent <br/> If no `login_hint` is supplied, sends `select_account` | N/A                  | Same as embedded flow                                                                                                                                                                                                                                                                                                  | (unsent - account selection via `AccountChooserActivity`) |
```
Comment (MariusVolkhart) [inline docs/prompt_behavior.md:14]: Probable typo: `preompted` -> prompted

**Comment by MariusVolkhart** [comment]
Perhaps library consumers are not the target audience, but it is unclear to me how the "Embedded Behavior" and "Broker Behavior" are observed. Broker behavior happens when a broker is available? Embedded is what happens when no broker is available?

**Comment by iambmelt** [comment]
> Perhaps library consumers are not the target audience, but it is unclear to me how the "Embedded Behavior" and "Broker Behavior" are observed. Broker behavior happens when a broker is available? Embedded is what happens when no broker is available?

Yep, that's exactly the distinction

File: docs/prompt_behavior.md (hunk 3 above)
Comment (iambmelt) [reply docs/prompt_behavior.md]: 7c077ed43ce1e18448c640d159df28c248caeecf

**Pull request merged**